src/core/ext/transport/chttp2/transport/internal.h

```diff
@@ -294,14 +295,13 @@ struct grpc_chttp2_transport {
   ~grpc_chttp2_transport();
 
   grpc_transport base; /* must be first */
+  gpr_mu mu;
```

Comment:
nit: I'd suggest using `grpc_core::Mutex` instead, and also using `grpc_core::MutexLock` and `grpc_core::ReleasableMutexLock` instead of `gpr_lock/unlock`.

Reply:
I would need to use ReleasableMutexLock everywhere and seems to add unnecessary wrappers and checks around it.. Any special reason for the switch?

Reply:
I know it's painful, but we have tried moving core to grpc_core and we shouldn't add new code using the C `gpr`.  For example, the init/destroy pay here should be done by the compiler.

As for `ReleasableMutexLock`, except a few places, most of those would use `grpc_core::MutexLock` with a simple block around the scope.

```
{  MutexLock lock(...)

}
```

In other places where you need `ReleasableMutexLock`, the branches are trimmed by the compiler because everything is inlined and can be inferred in simple cases.

Reply:
ah yeah, I can use scopes :)

Reply:
done!